host/xtest/pkcs11_1000.c

```diff
@@ -3741,3 +3741,428 @@ static void xtest_pkcs11_test_1016(ADBG_Case_t *c)
 }
 ADBG_CASE_DEFINE(pkcs11, 1016, xtest_pkcs11_test_1016,
 		 "PKCS11: Random number generator tests");
+
+static CK_RV derive_sym_key(CK_SESSION_HANDLE session,
+			    CK_OBJECT_HANDLE parent_key,
+			    CK_MECHANISM_TYPE mechanism, size_t data_len,
+			    CK_OBJECT_HANDLE_PTR derv_key_hdl, size_t keylen,
+			    CK_OBJECT_CLASS keyclass, CK_KEY_TYPE keytype,
+			    CK_BBOOL sensitive, CK_BBOOL extble)
+{
+	CK_RV rv = CKR_GENERAL_ERROR;
+	uint8_t buffer[516] = { 0 };
+	CK_MECHANISM mech_derive = { 0 };
+	CK_KEY_DERIVATION_STRING_DATA key_derv_param = { 0 };
+	CK_AES_CBC_ENCRYPT_DATA_PARAMS aes_cbc_param = { 0 };
+	CK_ATTRIBUTE derived_key_template[] = {
+		{ CKA_CLASS, &keyclass, sizeof(keyclass) },
+		{ CKA_KEY_TYPE, &keytype, sizeof(keytype) },
+		{ CKA_ENCRYPT, &(CK_BBOOL){CK_TRUE}, sizeof(CK_BBOOL) },
+		{ CKA_DECRYPT, &(CK_BBOOL){CK_TRUE}, sizeof(CK_BBOOL) },
+		{ CKA_SENSITIVE, &sensitive, sizeof(sensitive) },
+		{ CKA_EXTRACTABLE, &extble, sizeof(extble) },
+		{ CKA_VALUE_LEN, &keylen, sizeof(keylen) }
+	};
+
+	if (data_len > sizeof(buffer))
+		return rv;
+
+	memset(buffer, 0xCC, sizeof(buffer));
```

Comment:
you have data always 0xCC ?

Reply:
This was more for testing. I am varying the size of the buffer being used. For testing data wouldn't matter here.

Reply:
ok, then let's remove it